ui/app/components/project-input-variables/list-item.ts

```diff
@@ -0,0 +1,85 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { Variable } from 'waypoint-pb';
+import { inject as service } from '@ember/service';
+import ApiService from 'waypoint/services/api';
+
+interface VariableArgs {
+  variable: Variable.AsObject;
+  isEditing: boolean;
+  isCreating: boolean;
+  saveVariableSettings;
+}
+
+export default class ProjectInputVariablesListComponent extends Component<VariableArgs> {
+  initialVariable?: Variable.AsObject;
+  @service api!: ApiService;
+  @tracked args;
+  @tracked variable: Variable.AsObject;
+  @tracked isCreating: boolean;
+  @tracked isEditing: boolean;
+  @tracked isHcl: boolean;
+
+  constructor(owner: any, args: VariableArgs) {
+    super(owner, args);
+    this.isCreating = false;
+    let { variable, isEditing, isCreating } = args;
+    this.variable = variable;
+    this.isEditing = isEditing;
+    this.isCreating = isCreating;
+    this.isHcl = false;
+    if (variable.hcl) {
+      this.isHcl = true;
+    }
+    if (this.isEditing) {
+      this.initialVariable = JSON.parse(JSON.stringify(this.variable));
+    }
+  }
+
+  @action
+  async deleteVariable(variable) {
+    await this.args.deleteVariable(variable);
+  }
+
+  @action
+  editVariable(variable) {
+    this.isEditing = true;
+    this.initialVariable = JSON.parse(JSON.stringify(this.variable));
+  }
+
+  @action
+  async saveVariable(e) {
+    e.preventDefault();
+    await this.args.saveVariableSettings(this.variable);
+    this.isCreating = false;
+    this.isEditing = false;
+  }
+
+  @action
+  cancelCreate() {
+    this.isCreating = false;
+    this.isEditing = false;
+    this.deleteVariable(this.variable);
+  }
+
+  @action
+  cancelEdit() {
+    this.isCreating = false;
+    this.isEditing = false;
+    this.variable = this.initialVariable;
+  }
+
+  @action
+  toggleHcl(variable) {
+    if (this.isHcl) {
+      this.isHcl = false;
```

Comment:
Could we after the if/else statement do `this.isHcl = !this.isHcl` to reduce some of the similar logic here?

Reply:
I reworked this whole statement, so it's addressed